src/crystal/system/unix/event_libevent.cr

```diff
@@ -16,23 +18,19 @@ struct Crystal::Event
     @freed = false
   end
 
-  def add(timeout : LibC::Timeval? = nil)
+  def add(timeout : Time::Span?) : Nil
     if timeout
-      timeout_copy = timeout
-      LibEvent2.event_add(@event, pointerof(timeout_copy))
+      timeval = LibC::Timeval.new(
+        tv_sec: LibC::TimeT.new(timeout.total_seconds),
+        tv_usec: timeout.nanoseconds // 1_000
+      )
+      LibEvent2.event_add(@event, pointerof(timeval))
```

Comment:
Uh. Wait. Isn't `timeval` allocated on the stack here? Meaning that once `add` returns any pointer to it will be broken?

Reply:
It was previously already on the stack, I just removed the strange extra copy as it makes no sense. Just to be sure, I checked `event_add` source and it makes a copy right away (https://github.com/libevent/libevent/blob/8e03f4959736731975bf0eb0433e00486910ade6/event.c#L2638).